plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php

```diff
@@ -210,6 +210,29 @@ protected function get_order_item_data( $item ) {
 
 		// Expand meta_data to include user-friendly values.
 		$formatted_meta_data = $item->get_all_formatted_meta_data( null );
+
+		// Filter out product variations.
+		if ( isset( $product ) && 'true' === $this->request['order_item_display_meta'] ) {
+			$order_item_name   = $data['name'];
+			$data['meta_data'] = array_filter(
+				$data['meta_data'],
+				function( $meta ) use ( $product, $order_item_name ) {
+					$meta->key     = rawurldecode( (string) $meta->key );
+					$meta->value   = rawurldecode( (string) $meta->value );
+					$attribute_key = str_replace( 'attribute_', '', $meta->key );
+					$display_key   = wc_attribute_label( $attribute_key, $product );
+					$display_value = wp_kses_post( $meta->value );
+
+					// Skip items with values already in the product details area of the product name.
+					if ( $product && $product->is_type( 'variation' ) && wc_is_attribute_in_product_name( $display_value, $order_item_name ) ) {
```

Comment:
Before we double down on this, just wanted to flag [this issue](https://github.com/woocommerce/woocommerce/issues/28770) which was confirmed as a bug.

That issue refers to a scenario involving Woocommerce Product Addons, but you could also consider a case with nothing but WooCommerce running where we have a variable product named _"Backpack - Large"_ with a variation attribute _"Custom Logo Size"_ (where one of the options is _"Large")_ ... with this logic, we filter out potentially useful meta.

Reply:
I would say this is fine for now since it's consistent with core. When we fix it in core, it would be nice to fix this one as well. Ideally we'd have a single function that detects attributes that we can use in both places.